lib/wibox/layout/margin.lua

```diff
@@ -78,6 +78,14 @@ function margin:get_children()
     return {self.widget}
 end
 
+--- Replace the layout children
+-- This layout only accept one children, all others will be ignored
+-- @tparam table children A table composed of valid widgets
+function margin:set_children(children)
+    self.widget = children and children[1]
```

Comment:
So you are saying that (a) `check_widget()` should check if `.is_widget` is true (nice idea, you should add that :-) ) and (b) This doesn't count as public API and users shouldn't call it, right?